virttest/virsh.py

```diff
@@ -621,6 +624,93 @@ def kosher_args(remote_ip, uri):
         return True not in all_false
 
 
+class EventNotFoundError(Exception):
+    """
+    Error when certain event cannot be found.
+    """
+
+    def __init__(self, details=''):
```

Comment:
details= '' will hide silent failure where none of parameters is given __init__

Reply:
Ok, then what do you suggest we should put here?

Reply:
def __init__(self, details):
      self.details = details
In this way, if error will throw if no parameter is given when object  is constructed.
As zen of python says, Errors should never pass silently.

Reply:
@chunfuwen Of course, what do you suggest we put here? Seems you still didn't provide any suggestion?

Reply:
BTW, I didn't want to say this, but after second thought, maybe it could be helpful to you, so I'll just say it:
Seems you misunderstood the meaning of **Errors should never pass silently** which means we should always deal with errors, so....any more comments?

> def **init**(self, details):
> self.details = details
> In this way, error will throw if no parameter is given when object is constructed.
> As zen of python says, Errors should never pass silently.

Reply:
There is one assumption here that  you agree that obj = EventNotFoundError() is bad practice since there is no any meaning  input ,but object can created without any error. Otherwise there is no need to process this conversation. 
Below comments is based on above assumption.
```
if we define init as below, when you attempt to obj = EventNotFoundError() ,it will throw error immediately. That is part of zen of python says, Errors should never pass silently.
      def  __init__(self, details):
             self.details = details
```

Reply:
> There is one assumption here that you agree that obj = EventNotFoundError() is bad practice since there is no any meaning input ,but object can created without any error. Otherwise there is no need to process this conversation.

No I don't, so.....

> Below comments is based on above assumption.
> if we define init as below, when you attempt to obj = EventNotFoundError() ,it will throw error immediately. That is part of zen of python says, Errors should never pass silently.
> 
> ```
>       def  __init__(self, details):
>              self.details = details
> ```

So, any more comment?

Reply:
I just want to say again, since clearly you missed it, **Errors should never pass silently**  means we should always deal with errors.... not what you stated....

Reply:
> silently

I do believe "Errors should never pass silently" cover more than what you stated

Reply:
@chunfuwen of course, I can get that from your comments. We have different understanding about this sentence. But having thought about your suggestion, I'd keep current implementation.